gift-certificate-application/controller/src/main/java/com/epam/esm/controller/exception/AppAccessDeniedHandler.java

```diff
@@ -0,0 +1,29 @@
+package com.epam.esm.controller.exception;
+
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.web.access.AccessDeniedHandler;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+
+public class AppAccessDeniedHandler implements AccessDeniedHandler {
+    private static final String URI_STRING = "uri";
+    private static final String MSG_STRING = "msg";
+    private static final String TIMESTAMP_STRING = "timestamp";
+    private static final String ACCESS_DENIED_MSG = "Access denied";
+
+    @Override
+    public void handle(HttpServletRequest request,
+                       HttpServletResponse response,
+                       AccessDeniedException accessDeniedException) throws IOException {
+
+        HashMap<String, String> map = new HashMap<>();
+        map.put(URI_STRING, request.getRequestURI());
+        map.put(MSG_STRING, ACCESS_DENIED_MSG);
+        map.put(TIMESTAMP_STRING, String.valueOf(LocalDateTime.now()));
```

Comment:
These lines can be moved to a private method with a meaningful name describing what these lines do